src/solvers/interior_point_Newton.jl

```diff
@@ -0,0 +1,382 @@
+_doc_IPN = raw"""
+    interior_point_Newton(M, f, grad_f, Hess_f, p=rand(M); kwargs...)
+    interior_point_Newton(M, cmo::ConstrainedManifoldObjective, p=rand(M); kwargs...)
+    interior_point_Newton!(M, f, grad_f, Hess_f, p; kwargs...)
+    interior_point_Newton(M, ConstrainedManifoldObjective, p; kwargs...)
+
+perform the interior point Newton method following [LaiYoshise:2024](@cite).
+
+In order to solve the constrained problem
+
+```math
+\begin{aligned}
+\min_{p ∈\mathcal{M}} &f(p)\\
+\text{subject to } &g_i(p)\leq 0 \quad \text{ for } i= 1, …, m,\\
+\quad &h_j(p)=0 \quad \text{ for } j=1,…,n,
+\end{aligned}
+```
+
+This algorithms iteratively solves the linear system based on extending the KKT system
+by a slack variable `s`.
+
+```math
+\operatorname{J} F(p, μ, λ, s)[X, Y, Z, W] = -F(p, μ, λ, s),
+\text{ where }
+X ∈ T_p\mathcal M, Y,W ∈ ℝ^m, Z ∈ ℝ^n,
+```
+
+see [`CondensedKKTVectorFieldJacobian`](@ref) and [`CondensedKKTVectorField`](@ref), respectively,
+for the reduced form, this is usually solved in.
+From the resulting `X` and `Z` in the reeuced form, the other two, ``Y``, ``W``, are then computed.
+
+From the gradient ``(X,y,Z,W)`` at the current iterate ``(p, μ, λ, s)``,
```

Comment:
```suggestion
From the gradient ``(X,Y,Z,W)`` at the current iterate ``(p, μ, λ, s)``,
```
And this I think should be capital `Y`.